### DISCUSSION TITLE
Инициировать событие (event)

### DISCUSSION THREAD

**Discussion by Sanes** [moonshine-software/moonshine#30] [Q&A]
Можно подробней или пример, как работать с событиями? Observer не подходит.

**Comment by lee-to** [comment]
https://laravel.com/docs/9.x/eloquent#events

В документации есть множество примеров помимо Observers

**Reply by Sanes** [reply]
А конкретно в админке как это сделать?

**Reply by lee-to** [reply]
А причем тут админка, ресурс работает от модели, так и вы вешайте события к модели, как указано в документации

**Reply by lee-to** [reply]
Исходя из документации если вам требуется добавить событие на сохранение или удаление записи! 
`
protected $dispatchesEvents = [
    'saved' => UserSaved::class,
    'deleted' => UserDeleted::class,
];
`

**Reply by Sanes** [reply]
> Исходя из документации если вам требуется добавить событие на сохранение или удаление записи! `protected $dispatchesEvents = [ 'saved' => UserSaved::class, 'deleted' => UserDeleted::class, ];`

Тот же Observer получается. Придется проверять, откуда вызов. Из админской или клиентской части. Поэтому и хотел сделать на уровне метода контроллера у клиента, и ресурса у админа разные события.

**Reply by lee-to** [reply]
Добавил события в ресурс - https://moonshine.cutcode.ru/events-index
